File: src/scripts/read_excel.py
import zipfile
import os

file_path = r"C:\Users\tech solutionor\Downloads\CF PD Notes (1).xlsx"
print("File exists:", os.path.exists(file_path))
if os.path.exists(file_path):
    with zipfile.ZipFile(file_path, 'r') as zip_ref:
        print("Zip files:")
        for name in zip_ref.namelist():
            print(name)
            if 'sharedStrings' in name or 'sheet' in name:
                print(f"--- Content of {name} (first 800 chars) ---")
                with zip_ref.open(name) as f:
                    content = f.read(800)
                    print(content.decode('utf-8', errors='ignore'))
                    print("\n")

File: src/scripts/read_excel.js
const fs = require('fs');
const path = require('path');
const zlib = require('zlib');
const admZip = require('adm-zip'); // Let's see if we have adm-zip or if we can use another method

try {
  const filePath = 'C:\\Users\\tech solutionor\\Downloads\\CF PD Notes (1).xlsx';
  const zip = new admZip(filePath);
  const zipEntries = zip.getEntries();
  
  console.log("Zip Entries:");
  zipEntries.forEach(entry => {
    console.log(entry.entryName);
    if (entry.entryName.includes('sharedStrings') || entry.entryName.includes('sheet')) {
      const text = entry.getData().toString('utf8');
      console.log(`--- Content of ${entry.entryName} (first 500 chars) ---`);
      console.log(text.substring(0, 500));
    }
  });
} catch (err) {
  console.log("Error reading zip:", err.message);
}
